src/deluge/io/midi/midi_device.cpp

```diff
@@ -349,6 +353,51 @@ void MIDIDeviceUSB::sendMessage(uint8_t statusType, uint8_t channel, uint8_t dat
 	}
 }
 
+void MIDIDeviceUSB::sendSysex(uint8_t* data, int len) {
+	if (len < 3 || data[0] != 0xf0 || data[len - 1] != 0xf7) {
+		return;
+	}
+
+	int ip = 0;
+	ConnectedUSBMIDIDevice* connectedDevice = NULL;
+
+	for (int d = 0; d < MAX_NUM_USB_MIDI_DEVICES; d++) {
+		if (connectionFlags & (1 << d)) {
+			connectedDevice = &connectedUSBMIDIDevices[ip][d];
```

Comment:
I don't believe this works properly, won't it just send to the first connected device? Since each ConnectedUsbMidiDevice holds an array of 4 USBMidiDevices it should check that the current device is actually held by the connecteddevice it finds

Reply:
This follows the existing logic in sendMessage and it does seem to work. `connectionFlags` is private to the inner USBMIDIDevice and should only have a flag set for the specific connected device. It's a bit weird way of structuring things tho.

What I think we don't support at all is virtual port/cable differentiation for a specific connected device. That would be an important follow-up refactor.